test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -2249,7 +2254,11 @@ class test_ucp_sockaddr_protocols : public test_ucp_sockaddr {
             short_progress_loop();
             message = ucp_tag_probe_nb(receiver().worker(),
                                        0, 0, 1, &recv_info);
-        } while (message == NULL);
+        } while ((message == NULL) && !m_unreachable_err_count);
+
+        if ((message == NULL) && m_unreachable_err_count) {
```

Comment:
1. m_unreachable_err_count != 0
2. maybe no need to check message==NULL now?

Reply:
1. fixed
2. this check handles the case when RTS and unreachable error happens in the same progress loop, then need to complete the request in order to release `rdesc`

Reply:
ok